lib/library.rb

```diff
@@ -0,0 +1,60 @@
+class Library
+  attr_reader :name,
+              :library_books,
+              :authors,
+              :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @library_books = []
+    @authors = []
+    @checked_out_books = []
+  end
+
+  def add_author(author)
+    @authors << author
+    @library_books << author.books
+    @library_books.flatten
+  end
+
+  def publication_time_frame_for(author)
+    time_frame = {}
+    years = author.books.map do |book|
+      book.year.to_i
+    end
+    time_frame[:start] = years.min.to_s
+    time_frame[:end] = years.max.to_s
+    time_frame
+  end
+
+  def checkout(book)
+    if @library_books.include?(book)
+      @libaray_books.delete(book)
```

Comment:
oh, I see, you're deleting it. We'll want to keep our list of library_books intact so the library always knows which books it has, but then just add the book to the checked_out books.